packages/store/src/entity-base.ts

```diff
@@ -0,0 +1,228 @@
+import { StateStream } from './state-stream';
+import { getValue, setValue } from './utils';
+import { META_KEY } from './symbols';
+import { Injector } from '@angular/core';
+
+export interface EntityState<V> {
+  ids: (string | number)[];
+  entities: {
+    [id: string]: V;
+    [id: number]: V;
+  };
+}
+
+export type EntityUpdateStr<T> = {
+  id: string;
+  changes: Partial<T>;
+};
+
+export type EntityUpdateNum<T> = {
+  id: number;
+  changes: Partial<T>;
+};
+export type EntityUpdate<T> = EntityUpdateStr<T> | EntityUpdateNum<T>;
+
+export enum EntityMutation {
+  IdAndEntity,
+  Entity,
+  None
+}
+
+export abstract class EntityBase<T, S extends EntityState<T>> {
+  static defaults() {
+    return {
+      ids: [],
+      entities: {}
+    };
+  }
+
+  // @todo Don't like having to pass in an injector to be able to get at the
+  // state stream, how could we do this differently?
+  private stateStream: StateStream;
+  constructor(injector: Injector) {
```

Comment:
State stream really shouldn't be exposed. Its only like that for plugins. I don't think I like the idea of using it here.